Formula/lua-language-server.rb

```diff
@@ -0,0 +1,103 @@
+class LuaLanguageServer < Formula
+  desc "Language Server for the Lua language"
+  homepage "https://github.com/sumneko/lua-language-server"
+  # pull from git tag to get submodules
+  url "https://github.com/sumneko/lua-language-server.git",
+      tag:      "2.5.2",
+      revision: "cb2042160865589b5534a6bf0b6c366ae4ab1d99"
+  license "MIT"
+  head "https://github.com/sumneko/lua-language-server.git", branch: "master"
+
+  depends_on "ninja" => :build
+
+  # Disable `filesystem.test_appdata_path`.
+  #
+  # This test expeccts the application to run under the user's ${HOME},
+  # which is not compatible with homebrew's build environment.
+  # See https://github.com/actboy168/bee.lua/issues/21
+  # Adapted from https://github.com/macports/macports-ports/tree/master/lua/lua-language-server
+  patch :DATA
+
+  def install
+    system "git", "submodule", "update", "--init", "--recursive"
```

Comment:
Why can't it use the submodules as they are in the tag?

Reply:
You have to ask @sumneko -- those are the exact build instructions: https://github.com/sumneko/lua-language-server/wiki/Build-and-Run

Reply:
`brew` pulls submodules from Git clones automatically, so there's no need to do this.

Reply:
I trust you, but in a normal build this step is in fact necessary -- the build fails without it...

Reply:
Weird. Maybe `brew` doesn't do it recursively...

Reply:
I mean, it could be fine _on `brew`_; I just don't know... I am just going by my experience with the manual build. We can try, of course, and when it starts breaking down the line, add the line again?

Reply:
Ah, I see, yes. It should be fine. Let's try skipping this step in the formula.